lib/src/codepage.dart

```diff
@@ -277,6 +277,31 @@ CodePageDecoder _createDecoder(String characters) {
   return _NonBmpCodePageDecoder._(result);
 }
 
+/// An input [ByteConversionSink] for decoders where each input byte can be be
+/// considered independantly.
+class _CodePageDecoderSink implements ByteConversionSink {
+  final Sink<String> _output;
+  final String Function(List<int> input) _convert;
```

Comment:
I'd store the `Decoder` object itself here, and have a `_convert` method which calls `_decoder.convert`.
That avoids allocating an extra function closure.